### PR TITLE
Implemented new Jira connection with PAT authentication and plain jav…

### PR DESCRIPTION
**Pull request by JensHaed**
…a http API

### CODE REVIEW

**Comment by JensHaed** [comment]
Test seem to work but UI doesn't complete text. Further investigation shows that _CommonPrefixGrouper#getPossibleExpansions_ return null instead of an empty list. 
Text completion works if _CommonPrefixGrouper#getPossibleExpansions_ returns an empty list

File: src/main/kotlin/org/stt/connector/jira/JiraClient.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                .header("Accept", "application/json")
+                .header("Content-Type", "application/json")
+                .header("Authorization", "Basic " + Base64.getEncoder().encodeToString(("$username:$password").toByteArray()))
+                .build()
+        this.restClient?.send(request, HttpResponse.BodyHandlers.ofString())?.let {
+            if(it.statusCode() == 200) {
+                val deserializedAsMap = JsonIterator.deserialize(it.body(), Map::class.java)
+                val fields = deserializedAsMap["fields"] as Map<String, String>
+                return Issue(fields["summary"])
+            }
```
Comment (barclay-reg) [inline src/main/kotlin/org/stt/connector/jira/JiraClient.kt]: what about - at least some - error handling or logging here?

File: src/main/kotlin/org/stt/connector/jira/JiraClient.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        }
+                return Issue()
```
Comment (barclay-reg) [inline src/main/kotlin/org/stt/connector/jira/JiraClient.kt]: what kind of issue is this?

File: src/main/kotlin/org/stt/connector/jira/JiraConnector.kt [hunk 3]
```diff
@@ -11,25 +10,20 @@ import javax.inject.Singleton
...
 constructor(configuration: JiraConfig) : Service {
-//    private val client: JiraClient?
-
-//    private val projectNames: Set<String> by lazy {
-//        internalGetProjectNames()
-//    }
+    private val client: JiraClient?
 
     init {
-//        val jiraURI = configuration.jiraURI
```
Comment (barclay-reg) [inline src/main/kotlin/org/stt/connector/jira/JiraConnector.kt:21]: I'm also ok with deleting this code

File: src/test/kotlin/org/stt/text/JiraExpansionProviderTest.kt [hunk 4]
```diff
@@ -1,77 +1,77 @@
...
 import org.mockito.BDDMockito.given
 import org.mockito.Mock
 import org.mockito.MockitoAnnotations
+import org.stt.connector.jira.Issue
 import org.stt.connector.jira.JiraConnector
 import java.util.*
 
 class JiraExpansionProviderTest {
 
-//    @Mock
```
Comment (barclay-reg) [inline src/test/kotlin/org/stt/text/JiraExpansionProviderTest.kt:16]: please also here ... remove the commented-out code

File: src/main/kotlin/org/stt/text/JiraExpansionProvider.kt [hunk 5]
```diff
@@ -17,23 +17,21 @@ constructor(private val jiraConnector: JiraConnector,
             queryText = text.substring(spaceIndex, text.length).trim { it <= ' ' }
         }
 
-        return emptyList()
-
-//        return try {
```
Comment (barclay-reg) [inline src/main/kotlin/org/stt/text/JiraExpansionProvider.kt:22]: here, too - commented out code

**Review by barclay-reg — COMMENTED** [review]
I'm ok with it, if you say, that this is WorkInProgress ...

File: src/test/kotlin/org/stt/text/JiraExpansionProviderTest.kt (hunk 4 above)
Comment (JensHaed) [reply src/test/kotlin/org/stt/text/JiraExpansionProviderTest.kt:16]: It's the old version of the file. Tests comment-in and are adjusted to run successfully.

File: src/main/kotlin/org/stt/text/JiraExpansionProvider.kt (hunk 5 above)
Comment (JensHaed) [reply src/main/kotlin/org/stt/text/JiraExpansionProvider.kt:22]: It's the old version of the file. But comment-out code will be in use again.

File: src/main/kotlin/org/stt/connector/jira/JiraConnector.kt (hunk 3 above)
Comment (JensHaed) [reply src/main/kotlin/org/stt/connector/jira/JiraConnector.kt:21]: I'm not sure what you by "also ok with deleting", this code is replaced with a new implemention to connect to Jira

File: src/test/kotlin/org/stt/text/JiraExpansionProviderTest.kt (hunk 4 above)
Comment (barclay-reg) [reply src/test/kotlin/org/stt/text/JiraExpansionProviderTest.kt:16]: sorry for this - was confused

**Review by barclay-reg — APPROVED** [review]

**Comment by barclay-reg** [comment]
I'll leave to merge to you @JensHaed

**Pull request merged**